File: Python/Sorting.py
def bubbleSort(arr):
	n = len(arr)
	c=0
	while(c<n-1):
		for i in range(n-c-1):
			if arr[i]>arr[i+1]:
				arr[i],arr[i+1] = arr[i+1],arr[i]
		c+=1
	print(arr)

def selectionSort(arr):
 n=len(arr)
 for step in range(n):
  for i in range(step+1,n):
   if arr[step]>arr[i]:
   	arr[step] , arr[i] = arr[i] , arr[step]
 print (arr)

def insertionSort(arr):
	n=len(arr)							
	for i in range(n):
		key = arr[i]			
		j = i-1
		while (j>=0 and key < arr[j]):
			arr[j+1] = arr[j]
			j-=1
		arr[j+1]=key
	print(arr)

def mergeSort(arr):
	if len(arr)>1:
		n = len(arr)
		mid = n//2
		left = arr[:mid]
		right = arr[mid:]
		mergeSort(left)
		mergeSort(right)

		i=j=k=0

		while i<len(left) and j<len(right):
			if left[i]<right[j]:
				arr[k]=left[i]
				i+=1
			else:
				arr[k] = right[j]
				j+=1
			k+=1

		while i<len(left):
			arr[k]=left[i]
			k+=1
			i+=1

		while j<len(right):
			arr[k]=right[j]
			k+=1
			j+=1

def partition(arr ,l , h ):
	pivot = arr[l]
	i,j = l,h

	while i<j:

		while arr[i]<=pivot and i < h:
			i+=1

		while arr[j]>= pivot and j>l:
			j-=1
		if i<j : 
			arr[i],arr[j] = arr[j],arr[i]
		arr[l], arr[j] = arr[j],arr[l]
	return j

def quickSort(arr, l , h ):
	if l<h:
		pivot = partition(arr, l , h )

		quickSort(arr,l,pivot-1)
		quickSort(arr,pivot+1, h)


arr=[9,8,7,6,5,4,3,2,1]
# print('bubbleSort : ')
# bubbleSort(arr)
# print('selectionSort : ')
# selectionSort(arr
# insertionSort(arr)
# mergeSort(arr)
print(arr)
quickSort(arr,0,len(arr)-1)
print(arr)

File: CPP/sorting.cpp
#include<iostream>
#include<vector>
//#include<bits/stdc++.h>
using namespace std;

class Solution{
	
public:
	void BubbleSort(int arr[] , int n){
		cout<<"BUBBLE_SORT : ";
		for(int step=0 ; step<n-1 ; step++){
			for(int i=0 ; i<n-step-1; i++){
				if(arr[i] > arr[i+1]){
					int temp = arr[i];
					arr[i] = arr[i+1];
					arr[i+1] = temp;
				}
			}	
		}
	}

	void SelectionSort(int arr[] , int n ){
		cout<<"SELECTION SORT : ";
		for(int i=0;i<n-1;i++){
			for(int j=i+1;j<n;j++){
				if(arr[j]<arr[i]){
					int temp = arr[j];
					arr[j] = arr[i];
					arr[i] = temp;
				}
			}
		}
	}

	void insertionSort(int arr[] , int n){
		cout<<"INSERTION_SORT : ";
		for(int i=0;i<n;i++){
			int key = arr[i];
			int j=i-1;

			while(j>=0 and key < arr[j]){
				arr[j+1] = arr[j];
				j--;
			}
			arr[j+1] = key;
		}
	}

	void mergeSort(vector<int> &arr){
		int n = arr.size();
		if(n>1){
			int mid = n/2;

			vector<int> left;
			vector<int> right;

			for(int i=0;i<mid;i++)
				left.push_back(arr[i]);

			for(int i=mid;i<n;i++)
				right.push_back(arr[i]);

			this->mergeSort(left);
			this->mergeSort(right);

			int i=0,j=0,k=0;
			while(i<left.size() and j<right.size()){
				if(left[i]<right[j]){
					arr[k]=left[i];
					i++;
				}
				else{
					arr[k] = right[j];
					j++;
				}
				k++;
			}

			while( i<left.size()){
				arr[k] = left[i];
				k++;i++;
			}

			while(j<right.size()){
				arr[k] = right[j];
				k++;j++;
			}
		}
	}	
};

int main(){

	#ifndef ONLINE_JUDGE
		freopen("input.txt" , "r" , stdin);
		freopen("output.txt" , "w" , stdout);
	#endif

	//code goes here
	int n;cin>>n;
	int arr[n];
	for(int i=0;i<n;i++)
		cin>>arr[i];

	Solution obj;
	obj.BubbleSort(arr, n);
	for(auto i : arr)
		cout<<i<<" ";
	cout<<endl;

	cin>>n;
	int arr1[n];
	for(int i=0;i<n;i++){
		cin>>arr1[i];
	}

	// obj.SelectionSort(arr1, n);
	for(auto i : arr1)
		cout<<i<<" ";
	cout<<endl;

	// obj.insertionSort(arr1,n);
	int m;cin>>m;
	vector<int> array(m);
	for(int i=0;i<m;i++)
		cin>>array[i];

	for(int i=0;i<m;i++){
		cout<<array[i]<<" ";
	}
	cout<<"MERGESORT : ";
	obj.mergeSort(array);

	for(int i=0;i<m;i++){
		cout<<array[i]<<" ";
	}

return 0;
}